billmategateway/controllers/front/accept.php

```diff
@@ -60,297 +60,297 @@ public function getmoduleId($method)
+                        else
+                            $order_id = Order::getOrderByCartId($this->context->cart->id);
+
+                        $orderObject = new Order($order_id);
+                        if ($orderObject->current_state == Configuration::get('BILLMATE_PAYMENT_PENDING') && $data['status'] != 'Pending') {
+                            $orderHistory = new OrderHistory();
+
+                            $status_key = 'B' . strtoupper($this->method) . '_ORDER_STATUS';
+                            if ($this->method == 'checkout') {
+                                $status_key = 'BILLMATE_CHECKOUT_ORDER_STATUS';
+                            }
 
-						$status = ($data['status'] == 'Cancelled') ? Configuration::get('PS_OS_CANCELED') : $status;
-						$orderHistory->id_order = (int) $orderObject->id;
-						$orderHistory->changeIdOrderState($status,(int) $orderObject->id, true);
-						$orderHistory->add();
-					}
-					if(isset($this->context->cookie->billmatepno))
-						unset($this->context->cookie->billmatepno);
+                            $status = Configuration::get($status_key);
 
-                    if ('' != Common::getCartCheckoutHash()) {
-                        $hash = Common::getCartCheckoutHash();
-                        Common::unsetCartCheckoutHash();
-						$url = $this->context->link->getModuleLink(
-							'billmategateway',
-							'thankyou',
-							array('billmate_hash' => $hash));
-						Tools::redirectLink($url);
-						die;
-					} else {
-
-
-						Tools::redirectLink(__PS_BASE_URI__ . 'order-confirmation.php?key=' . $customer->secure_key .
-							'&id_cart=' . (int)$this->context->cart->id . '&id_module=' . (int)$this->getmoduleId('billmate' . $this->method) .
-							'&id_order=' . (int)$order_id);
-						die;
-					}
-				} else {
-
-
-                    if ($paymentInfo['PaymentData']['method'] == '1' OR '2' == $paymentInfo['PaymentData']['method']) {
-                        /** Paid with invoice */
-                        $this->paymentClass = 'invoice';
-                        $class_file = _PS_MODULE_DIR_.'billmategateway/methods/'.Tools::ucfirst($this->paymentClass).'.php';
-                        require_once($class_file);
-                        $class        = "BillmateMethod".Tools::ucfirst($this->paymentClass);
-                        $this->module = new $class;
-                    }
+                            $status = ($data['status'] == 'Cancelled') ? Configuration::get('PS_OS_CANCELED') : $status;
+                            $orderHistory->id_order = (int)$orderObject->id;
+                            $orderHistory->changeIdOrderState($status, (int)$orderObject->id, true);
+                            $orderHistory->add();
+                        }
+                        if (isset($this->context->cookie->billmatepno))
+                            unset($this->context->cookie->billmatepno);
+
+                        if ('' != Common::getCartCheckoutHash()) {
```

Comment:
Avoid Yoda conditions, should be: 

`if (Common::getCartCheckoutHash() != '') {`